pandas/core/arrays/arrow/array.py

```diff
@@ -2088,6 +2091,62 @@ def _dt_isocalendar(self):
     def _dt_is_leap_year(self):
         return type(self)(pc.is_leap_year(self._pa_array))
 
+    @property
+    def _dt_is_month_start(self):
+        return type(self)(pc.equal(pc.day(self._pa_array), 1))
+
+    @property
+    def _dt_is_month_end(self):
+        plus_one_day = pc.add(self._pa_array, pa.scalar(datetime.timedelta(days=1)))
+        return type(self)(pc.equal(pc.day(plus_one_day), 1))
+
+    @property
+    def _dt_is_year_start(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 1),
+                pc.equal(pc.day(self._pa_array), 1),
+            )
+        )
+
+    @property
+    def _dt_is_year_end(self):
+        return type(self)(
+            pc.and_(
+                pc.equal(pc.month(self._pa_array), 12),
+                pc.equal(pc.day(self._pa_array), 31),
+            )
+        )
+
+    @property
+    def _dt_is_quarter_start(self):
+        is_correct_month = pc.is_in(pc.month(self._pa_array), pa.array([1, 4, 7, 10]))
+        is_first_day = pc.equal(pc.day(self._pa_array), 1)
+        return type(self)(pc.and_(is_correct_month, is_first_day))
```

Comment:
Could you use [`pc.floor_temporal`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.quarter.html#pyarrow.compute.quarter)? E.g:
```python
return pc.equal(pc.floor_temporal(self._pa_array, unit='quarter'), self._pa_array)
```

Reply:
Got it. How about:
```python
pc.equal(pc.floor_temporal(self._pa_array, unit='quarter'), pc.floor_temporal(self._pa_array, unit='day'))
```